Importing ibc-testkit failed

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺ 
v               ✰  Thanks for opening an issue! ✰    
v    Before smashing the submit button please review the template.
v    Please also ensure that this is not a duplicate issue :)  
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Bug Summary
I tried importing `ibc-testkit` and failed due to `error: failed to select a version for ibc-app-transfer.`

```
ibc = {version = "0.48.0", default-features = false, features = ["serde"]}
ibc-derive = "0.4.0"
ibc-proto = {version = "0.38.0", default-features = false}
ibc-testkit = {version = "0.48.0", default-features = false, optional = true}
```

## Details

```
error: failed to select a version for `ibc-app-transfer`.
    ... required by package `ibc-apps v0.48.0`
    ... which satisfies dependency `ibc-apps = "^0.48.0"` of package `ibc v0.48.0`
    ... which satisfies dependency `ibc = "^0.48.0"` of package `ibc-testkit v0.48.0`
    ... which satisfies dependency `ibc-testkit = "^0.48.0"` of package `namada_core v0.27.0 (/Users/yuji/Workspace/anoma/namada/core)`
    ... which satisfies path dependency `namada_core` of package `namada v0.27.0 (/Users/yuji/Workspace/anoma/namada/shared)`
    ... which satisfies path dependency `namada` of package `namada_apps v0.27.0 (/Users/yuji/Workspace/anoma/namada/apps)`
    ... which satisfies path dependency `namada_apps` of package `namada_benchmarks v0.27.0 (/Users/yuji/Workspace/anoma/namada/benches)`
versions that meet the requirements `^0.48.0` are: 0.48.0

the package `ibc-apps` depends on `ibc-app-transfer`, with features: `serde_json` but `ibc-app-transfer` does not have these features.
 It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.


failed to select a version for `ibc-app-transfer` which could resolve this conflict
make: *** [build] Error 101
```

I tried the following change, it worked.

```diff
diff --git a/ibc-apps/ics20-transfer/Cargo.toml b/ibc-apps/ics20-transfer/Cargo.toml
index 2e992e71..4d54f2bf 100644
--- a/ibc-apps/ics20-transfer/Cargo.toml
+++ b/ibc-apps/ics20-transfer/Cargo.toml
@@ -38,7 +38,7 @@ std = [
 serde = [
     "ibc-app-transfer-types/serde",
     "ibc-core/serde",
-    "dep:serde_json"
+    "serde_json"
 ]
 schema = [
     "ibc-app-transfer-types/schema",
```

## Version

`v0.48.0`

--- a/.changelog/unreleased/bug-fixes/987-serde-json-feature.md
+++ b/.changelog/unreleased/bug-fixes/987-serde-json-feature.md
@@ -1,0 +1,2 @@
+- Fix dependency resolution whan building no_std
+  ([#987](https://github.com/cosmos/ibc-rs/issues/987))

--- a/.changelog/unreleased/bug-fixes/987-serde-json-feature.md
+++ b/.changelog/unreleased/bug-fixes/987-serde-json-feature.md
@@ -1,2 +1,3 @@
-- Fix dependency resolution whan building no_std
+- Fix dependency resolution by removing the `dep:` syntax in `serde` feature of
+  `ibc-app-transfer` crate.
   ([#987](https://github.com/cosmos/ibc-rs/issues/987))

--- a/ibc-apps/ics20-transfer/Cargo.toml
+++ b/ibc-apps/ics20-transfer/Cargo.toml
@@ -33,12 +33,12 @@ default = ["std"]
 std = [
     "ibc-app-transfer-types/std",
     "ibc-core/std",
-    "serde_json?/std",
+    "serde_json/std",
 ]
 serde = [
     "ibc-app-transfer-types/serde",
     "ibc-core/serde",
-    "dep:serde_json"
+    "serde_json"
 ]
 schema = [
     "ibc-app-transfer-types/schema",

--- a/ibc-apps/ics20-transfer/Cargo.toml
+++ b/ibc-apps/ics20-transfer/Cargo.toml
@@ -33,7 +33,7 @@ default = ["std"]
 std = [
     "ibc-app-transfer-types/std",
     "ibc-core/std",
-    "serde_json/std",
+    "serde_json?/std",
 ]
 serde = [
     "ibc-app-transfer-types/serde",

